pytorch_lightning/trainer/trainer.py

```diff
@@ -1105,6 +1105,14 @@ def call_configure_sharded_model(self, model: LightningModule) -> None:
             model.call_configure_sharded_model_hook = True
             self.accelerator.call_configure_sharded_model_hook = False
 
+    @property
+    def sharded_module(self) -> torch.nn.Module:
+        """
+        Returns: The Sharded version of the ``LightningModule`` module if using sharded,
+            else the model stored in the accelerator.
+        """
+        return self.accelerator.sharded_module
```

Comment:
Would it make sense to do an instance check here for the plugin and call sharded_module on the plugin? Instead of having a specific sharded_module property on the accelerator.

Reply:
This would mean the trainer is aware of the training type plugin, hence routing it through the accelerator